esp/services/ws_dali/ws_daliservice.cpp

```diff
@@ -0,0 +1,103 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifdef _USE_OPENLDAP
+#include "ldapsecurity.ipp"
+#endif
+
+#include "ws_daliservice.hpp"
+#include "jlib.hpp"
+#include "dautils.hpp"
+#include "dasds.hpp"
+
+#define REQPATH_EXPORTSDSDATA "/WSDali/Export"
+
+const char* daliFolder = "tempdalifiles" PATHSEPSTR;
+
+void CWSDaliEx::init(IPropertyTree* cfg, const char* process, const char* service)
+{
+    espProcess.set(process);
+}
+
+int CWSDaliSoapBindingEx::onGet(CHttpRequest* request, CHttpResponse* response)
+{
+    try
+    {
+#ifdef _USE_OPENLDAP
+        request->queryContext()->ensureSuperUser(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Access denied, administrators only.");
+#endif
+        if (wsdService->isDaliDetached())
+            throw makeStringException(ECLWATCH_CANNOT_CONNECT_DALI, "Dali detached.");
+
+        StringBuffer path;
+        request->getPath(path);
+
+        if (!strnicmp(path.str(), REQPATH_EXPORTSDSDATA, sizeof(REQPATH_EXPORTSDSDATA) - 1))
+        {
+            exportSDSData(request, response);
+            return 0;
+        }
+    }
+    catch(IException* e)
+    {
+        onGetException(*request->queryContext(), request, response, *e);
+        FORWARDEXCEPTION(*request->queryContext(), e,  ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return CWSDaliSoapBinding::onGet(request,response);
+}
+
+void CWSDaliSoapBindingEx::exportSDSData(CHttpRequest* request, CHttpResponse* response)
+{
+    CDateTime now;
+    now.setNow();
+    time_t createTime = now.getSimple();
+
+    StringBuffer peer, outFileNameWithPath;
+    VStringBuffer outFileName("sds_for_%s_%ld", request->getPeer(peer).str(), createTime);
```

Comment:
If using a file you could consider using the function 
```
extern jlib_decl IFileIO *createUniqueFile(const char *dir, const char *prefix, const char *ext, StringBuffer &tmpName);
```
or similar.  To make sure your file is unque.